sys/src/9/ufs/ufs_harvey.h

```diff
@@ -9,19 +9,52 @@
  */
 
 
-typedef	struct Chan	Chan;
+typedef struct Chan Chan;
+typedef struct ufsmount ufsmount;
+typedef struct vnode vnode;
+typedef struct thread thread;
+typedef struct inode inode;
 
 
-struct mount {
-	struct ufsmount*	mnt_data;
-	Chan*			chan;
-};
+/* Wrapper for a UFS mount.  Should support reading from both kernel and user
+ * space (eventually)
+ */
+typedef struct MountPoint {
+	ufsmount	*mnt_data;
+	Chan		*chan;
+} MountPoint;
+
 
 /* Harvey equivalent to FreeBSD vnode, but not exactly the same.  Acts as a
  * wrapper for the inode and any associated data.  This is not intended to be
  * support multiple filesystems and should probably be renamed after it works.
  */
-struct vnode {
-	struct inode*	v_data;
-	//struct mount*	v_mount;
-};
+typedef struct vnode {
+	inode	*v_data;
+	//MountPoint	*v_mount;
+} vnode;
+
+
+// Not sure we even need this - if not we can remove it later.
+typedef struct thread {
```

Comment:
it doesn't make a lot of sense in the kernel ... in harvey, there's no such thing as a kernel level thread. Just processes.